evalml/tests/component_tests/test_estimators.py

```diff
@@ -441,3 +444,49 @@ def test_estimator_prediction_intervals(
     assert (predictions > result_75_85["0.75_lower"]).all()
     assert (result_75_85["0.85_lower"] > result_95["0.95_lower"]).all()
     assert (result_75_85["0.75_lower"] > result_75_85["0.85_lower"]).all()
+
+
+@pytest.mark.parametrize(
+    "incompatible_estimator",
+    [
+        comp
+        for comp in _all_estimators()
+        if comp._integer_nullable_incompatibilities
+        or comp._boolean_nullable_incompatibilities
+    ],
+)
+def test_estimators_with_nullable_type_incompatibilities(
+    split_nullable_logical_types_by_compatibility,
+    nullable_type_test_data,
+    nullable_type_target,
+    incompatible_estimator,
+):
+    if incompatible_estimator == ARIMARegressor:
+        pytest.skip(
+            f"Skipping because {incompatible_estimator} has nullable type handling in place to cover its errors",
+        )
+
+    (_, incompatible_y_ltypes) = split_nullable_logical_types_by_compatibility(
+        "y" in incompatible_estimator._integer_nullable_incompatibilities,
+        "y" in incompatible_estimator._boolean_nullable_incompatibilities,
+    )
+
+    for y_ltype in incompatible_y_ltypes:
+        # Most estimators cannot handle nans
+        y = nullable_type_target(ltype=y_ltype, has_nans=False)
+        X = nullable_type_test_data(has_nans=False)
+
+        # Exclude logical types that most estimators cannot handle
+        X = X.ww.select(["numeric", "Boolean", "BooleanNullable"])
+
+        comp = incompatible_estimator()
+
+        with pytest.raises(Exception):
+            comp.fit(X, y)
+            comp.predict(X)
```

Comment:
Because we are using the actaul evalml components here, this will have to change once we integrate into automl search to call the underlying `component_obj` so that we can track when support gets added for that component.